src/protocols/telnet/user.c

```diff
@@ -92,7 +92,7 @@ int guac_telnet_user_join_handler(guac_user* user, int argc, char** argv) {
         user->pipe_handler = guac_telnet_pipe_handler;
 
         /* Updates to connection parameters */
-        user->argv_handler = guac_telnet_argv_handler;
+        user->argv_handler = guac_argv_handler;
```

Comment:
Looks like this is consistently changed to the `guac_argv_handler` for all of the protocols - does it make sense to continue to allow this to be configurable, or is it something that should be changed in the `guac_user` code such that `guac_argv_handler` is always what is called?

I'm good with it either way, just curious - I'm all for leaving things flexible/configurable, just wasn't sure if the new `argv` code is intended to be the new flexible/configurable framework that is always referred to, or if there are foreseeable situations where you'd still want to change that?

Reply:
I definitely think that `argv_handler` needs to stay there:

The pattern of the API provided by libguac is one of gradually increasing levels of abstraction, and I think it's important that we maintain that pattern consistently. We provide higher-level convenience functions, default implementations, etc. as needed, but aim to always allow implementations to take control if those conveniences aren't well-suited in their case.

Regarding this specific implementation, it provides its convenience at the cost of limited buffer space for received argument values. Should an implementation need greater space, or to process arguments of any size as true streams, they will need to be able to handle the raw stream themselves.